From 773a5ce096a1b3d411b5e72d895729007d2cb08c Mon Sep 17 00:00:00 2001
From: Wang Yufen <wangyufen@huawei.com>
Date: Tue, 7 Feb 2023 17:07:58 +0800
Subject: [PATCH] kabi: add early_demux_handler and early_demux back to fix
 kabi broken in struct net_protocol and inet6_protocol
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 773a5ce096a1b3d411b5e72d895729007d2cb08c
Modified-by-SEL: No


Offering: HULK
hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I6D0ZE

--------------------------------

Add early_demux_handler and early_demux back to fix kabi broken in
struct net_protocol and inet6_protocol.

Signed-off-by: Wang Yufen <wangyufen@huawei.com>
Reviewed-by: Liu Jian <liujian56@huawei.com>
Signed-off-by: Jialin Zhang <zhangjialin11@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/net/protocol.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/net/protocol.h b/include/net/protocol.h
index 0fd2df844fc7..2b778e1d2d8f 100644
--- a/include/net/protocol.h
+++ b/include/net/protocol.h
@@ -35,6 +35,8 @@
 
 /* This is used to register protocols. */
 struct net_protocol {
+	int			(*early_demux)(struct sk_buff *skb);
+	int			(*early_demux_handler)(struct sk_buff *skb);
 	int			(*handler)(struct sk_buff *skb);
 
 	/* This returns an error if we weren't able to handle the error. */
@@ -51,6 +53,8 @@ struct net_protocol {
 
 #if IS_ENABLED(CONFIG_IPV6)
 struct inet6_protocol {
+	void	(*early_demux)(struct sk_buff *skb);
+	void    (*early_demux_handler)(struct sk_buff *skb);
 	int	(*handler)(struct sk_buff *skb);
 
 	/* This returns an error if we weren't able to handle the error. */
-- 
2.33.0

